### PR TITLE
Print decompiled refinements

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 1]
```diff
@@ -859,27 +888,15 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
           case _ => this
         }
 
+      case Type.MethodType(paramNames, params, body) =>
+        this += "("
+        printMethodicTypeParams(paramNames, params)
+        this += ") => "
+        printTypeOrBound(body)
+
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: Why is this necessary? I assume method types are handled already in refinement case.

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 2]
```diff
@@ -74,6 +74,7 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
 
         if (flags.isObject) this += "object " += name.stripSuffix("$")
         else if (flags.isTrait) this += "trait " += name
+        else if (flags.isAbstract) this += "abstract class " += name
         else this += "class " += name
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: `final` is a few lines above. I will add `sealed `.

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 3]
```diff
@@ -793,12 +779,14 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
             printType(prefix)
             this += "#"
-          case prefix@Type() =>
-            printType(prefix)
-            this += "."
+          case prefix @ Type() =>
+            if (!sym.flags.isLocal) {
+              printType(prefix)
+              this += "."
+            }
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: This is to not print prefixes in references inside the parents of a class.

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 4]
```diff
@@ -822,9 +810,50 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
+                      this += "]"
+                      printMethodicType(res)
+                    case Type.ByNameType(t) =>
+                      this += ": "
+                      printType(t)
+                    case tp @ Type() =>
+                      this += ": "
+                      printType(tp)
+                  }
+                  printMethodicType(info)
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: Will refactor

File: library/src/scala/tasty/util/ShowSourceCode.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: As far as I can see we do not need it. I will remove it.

File: library/src/scala/tasty/util/ShowSourceCode.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: I will reprint the prefix and add test cases where this fails
